source/common/stats/histogram_impl.h

```diff
@@ -15,18 +17,38 @@
 namespace Envoy {
 namespace Stats {
 
+class HistogramSettingsImpl : public HistogramSettings {
+public:
+  HistogramSettingsImpl() = default;
+  HistogramSettingsImpl(const envoy::config::metrics::v3::StatsConfig& config);
+
+  // HistogramSettings
+  const ConstSupportedBuckets& buckets(absl::string_view stat_name) const override;
+
+  static ConstSupportedBuckets& defaultBuckets();
+
+private:
+  using Config = std::pair<Matchers::StringMatcherImpl, ConstSupportedBuckets>;
+  const std::vector<Config> configs_{};
```

Comment:
I think the init is not necessary?

Reply:
This was needed to fix the build in gcc. It complained that with the default constructor, this const member was uninitialized. I think gcc was wrong in this case, because std::vector has a sane default constructor also (which set it to the value I wanted it to have in that case), but alas, I don't write the compilers.